### PR TITLE
Updates to Ubuntu build nodes

### PR DESCRIPTION
**Pull request by spham-amzn**
- Fix version string for cmake to allow incremental updates of cmake at kitware's ubuntu repo
- Add required libfontconfig1-dev which is needed by Linux/Qt/QtFontDatabaseSupport (dependency of Qt5XcbQpa)

### CODE REVIEW

**Review by lumberyard-employee-dm — APPROVED** [review]

**Comment by lumberyard-employee-dm** [comment]
Do you mean to to merge this to development?

**Review by amzn-changml — APPROVED** [review]
Assuming this is being merged to the correct destination branch, this seems reasonable. 

Normally not a fan of unbounded versions, but we can roll back AMIs if it comes to it

**Comment by spham-amzn** [comment]
> 
> 
> Assuming this is being merged to the correct destination branch, this seems reasonable.
> 
> Normally not a fan of unbounded versions, but we can roll back AMIs if it comes to it

Yeah, but it appears that they dont keep older versions available through their apt repo, but i didnt look too deep into it

**Comment by spham-amzn** [comment]
> 
> 
> Do you mean to to merge this to development?

Yes

File: scripts/build/build_node/Platform/Linux/package-list.ubuntu-bionic.txt [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-cmake/3.20.1-0kitware1ubuntu18.04.1     # For cmake
+cmake/3.20.*                            # For cmake
 clang-6.0                               # For Ninja Build System
 ninja-build                             # For the compiler and its dependencies
 
 # Build Libraries
 libglu1-mesa-dev                        # For Qt (GL dependency)
 libxcb-xinerama0                        # For Qt plugins at runtime
 libxcb-xinput0                          # For Qt plugins at runtime
+libfontconfig1-dev						# For Qt plugins at runtime
```
Comment (amznestebanpapp) [inline scripts/build/build_node/Platform/Linux/package-list.ubuntu-bionic.txt:12]: nit: can you fix the identation of the comment?

File: scripts/build/build_node/Platform/Linux/package-list.ubuntu-focal.txt [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
-cmake/3.20.1-0kitware1ubuntu20.04.1     # For cmake
+cmake/3.20.*                            # For cmake
 clang-6.0                               # For Ninja Build System
 ninja-build                             # For the compiler and its dependencies
 
 # Build Libraries
 libglu1-mesa-dev                        # For Qt (GL dependency)
 libxcb-xinerama0                        # For Qt plugins at runtime
 libxcb-xinput0                          # For Qt plugins at runtime
+libfontconfig1-dev						# For Qt plugins at runtime
```
Comment (amznestebanpapp) [inline scripts/build/build_node/Platform/Linux/package-list.ubuntu-focal.txt:12]: same here

**Review by amznestebanpapp — APPROVED** [review]

**Comment by amznestebanpapp** [comment]
> > Assuming this is being merged to the correct destination branch, this seems reasonable.
> > Normally not a fan of unbounded versions, but we can roll back AMIs if it comes to it
> 
> Yeah, but it appears that they dont keep older versions available through their apt repo, but I didnt look too deep into it

I am with @amzn-mike on that. If you need to re-generate the AMI and the version is no longer available, I would rather have it fail than not realizing some version was updated. We usually need to run some validations to be able to update those versions.

CMake patch versions have caused issues (3.20.4 with VS19.9), so you could end up in that situation. We also have things like a check in the generation of the installer that will validate the exact CMake version being used. We dont have this for Linux, but when we do, it could cause issues there.

It also sets a bad example for other dependencies.

**Comment by spham-amzn** [comment]
> 
> 
> > > Assuming this is being merged to the correct destination branch, this seems reasonable.
> > > Normally not a fan of unbounded versions, but we can roll back AMIs if it comes to it
> > 
> > 
> > Yeah, but it appears that they dont keep older versions available through their apt repo, but I didnt look too deep into it
> 
> I am with @amzn-mike on that. If you need to re-generate the AMI and the version is no longer available, I would rather have it fail than not realizing some version was updated. We usually need to run some validations to be able to update those versions.
> 
> CMake patch versions have caused issues (3.20.4 with VS19.9), so you could end up in that situation. We also have things like a check in the generation of the installer that will validate the exact CMake version being used. We dont have this for Linux, but when we do, it could cause issues there.
> 
> It also sets a bad example for other dependencies.

This was done only because the specific version was removed, not because we want the latest. The script as is will fail to install any cmake

**Comment by spham-amzn** [comment]
Discarding in favor of https://github.com/o3de/o3de/pull/1985 (due to DCO issues from the development branch)

**Pull request closed without merging**